FilePersistence/src/version_control/ListMergeComponent.cpp

```diff
@@ -172,17 +172,29 @@ LinkedChangesTransition ListMergeComponent::run(std::shared_ptr<GenericTree>& tr
 				for (auto elemId : chunk->spanBase_)
 				{
 					auto oldNode = treeBase->find(elemId);
-					auto newNode = treeA->find(elemId);
+					auto newNodeA = treeA->find(elemId);
+					auto newNodeB = treeB->find(elemId);
 					Q_ASSERT(oldNode);
-					Q_ASSERT(newNode);
-					auto newChange = std::make_shared<ChangeDescription>(oldNode, newNode);
-					updateChange(newChange, cdgA, cdgB);
-					newChange->print();
-					auto delChange = cdgA.changes().value(elemId);
-					cdgA.remove(delChange);		// Remove previous change
-					cdgA.insert(newChange);
-					cdgA.recordDependencies(newChange, true);
-					cdgA.recordDependencies(newChange, false);
+					Q_ASSERT(newNodeA || newNodeB);
+					if (newNodeA)	//Add change in cdgA
+					{
+						auto newChange = std::make_shared<ChangeDescription>(oldNode, newNodeA);
```

Comment:
Instead of repeating almost identical code twice, make a lambda and call with with appropriate arguments.